runtime/src/vote_account.rs

```diff
@@ -172,12 +174,27 @@ impl<'de> Deserialize<'de> for ArcVoteAccount {
     }
 }
 
+impl From<AccountSharedData> for ArcVoteAccount {
+    fn from(account: AccountSharedData) -> Self {
+        Self(Arc::new(VoteAccount::from(account)))
+    }
+}
 impl From<Account> for ArcVoteAccount {
     fn from(account: Account) -> Self {
         Self(Arc::new(VoteAccount::from(account)))
     }
 }
 
+impl From<AccountSharedData> for VoteAccount {
```

Comment:
@behzadnouri does this look ok now?

Reply:
the issue was changing `Serialize/Deserialize` impl which is now reverted.
changes to `new_rand_vote_account` also seem redundant but i guess it does not make a difference.